File: setMatrixZeros.js
function setZeroes(matrix){
    let rows=[];
    let cols=[];
    const n=matrix.length;
    const m=matrix[0].length;
    
    for(let i =0;i<n;i++){
        for(let j=0;j<m;j++){
            if(matrix[i][j]===0){
                rows.push(i);
                cols.push(j);
            }
        }
    }

    for(let i of rows){
        for(let j=0;j<m;j++){
            matrix[i][j]=0;
        }
    }

    for(let j of cols){
        for(let i=0;j<n;j++){
            matrix[i][j]=0;
        }
    }
}


function main() {
    let mat = [[1,1,1],[1,0,1],[1,1,1]];
    setZeroes(mat);
    console.log(mat); // Should print [[1,0,1],[0,0,0],[1,0,1]]
}

main();

File: setMatrixZeros.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   public:
    void setZeroes(vector<vector<int>>& matrix) {
        int row = matrix.size();
        int col = matrix[0].size();

        vector<int> rows(row, 1);
        vector<int> cols(col, 1);

        for (int rc = 0; rc < row; rc++) {
            for (int cc = 0; cc < col; cc++) {
                if (matrix[rc][cc] == 0) {
                    rows[rc] = 0;
                    cols[cc] = 0;
                }
            }
        }

        for (int rc = 0; rc < row; rc++) {
            for (int cc = 0; cc < col; cc++) {
                if (rows[rc] == 0 || cols[cc] == 0) {
                    matrix[rc][cc] = 0;
                }
            }
        }
    }
};

int main() {
    return 0;
}
